### PR TITLE
[WIP] Update MySQL JDBC driver

### LINKED ISSUE #443: MySQL connector prints NPE during SELECT
It happens when I executed a simple SELECT query to mysql on presto-product-tests. The returned result is fine, but NPE message is printed in the server log. 
```sql
presto:sys> select * from mysql.sys.version;
 sys_version | mysql_version 
-------------+---------------
 1.5.1       | 5.7.25        
```
Even if the schema isn't sys, the same message appears.

```sh
2019-03-11T18:14:16.377+0545	INFO	Query-20190311_122916_00007_esdqa-167	stderr	Mon Mar 11 18:14:16 NPT 2019 WARN: Establishing SSL connection without server's identity verification is not recommended. According to MySQL 5.5.45+, 5.6.26+ and 5.7.6+ requirements SSL connection must be established by default if explicit option isn't set. For compliance with existing applications not using SSL the verifyServerCertificate property is set to 'false'. You need either to explicitly disable SSL by setting useSSL=false, or set useSSL=true and provide truststore for server certificate verification.
2019-03-11T18:14:16.428+0545	INFO	Query-20190311_122916_00007_esdqa-167	stderr	Mon Mar 11 18:14:16 NPT 2019 WARN: Establishing SSL connection without server's identity verification is not recommended. According to MySQL 5.5.45+, 5.6.26+ and 5.7.6+ requirements SSL connection must be established by default if explicit option isn't set. For compliance with existing applications not using SSL the verifyServerCertificate property is set to 'false'. You need either to explicitly disable SSL by setting useSSL=false, or set useSSL=true and provide truststore for server certificate verification.
2019-03-11T18:14:16.474+0545	INFO	Query-20190311_122916_00007_esdqa-167	stderr	Mon Mar 11 18:14:16 NPT 2019 WARN: Establishing SSL connection without server's identity verification is not recommended. According to MySQL 5.5.45+, 5.6.26+ and 5.7.6+ requirements SSL connection must be established by default if explicit option isn't set. For compliance with existing applications not using SSL the verifyServerCertificate property is set to 'false'. You need either to explicitly disable SSL by setting useSSL=false, or set useSSL=true and provide truststore for server certificate verification.
2019-03-11T18:14:16.547+0545	INFO	Query-20190311_122916_00007_esdqa-172	stderr	Mon Mar 11 18:14:16 NPT 2019 WARN: Establishing SSL connection without server's identity verification is not recommended. According to MySQL 5.5.45+, 5.6.26+ and 5.7.6+ requirements SSL connection must be established by default if explicit option isn't set. For compliance with existing applications not using SSL the verifyServerCertificate property is set to 'false'. You need either to explicitly disable SSL by setting useSSL=false, or set useSSL=true and provide truststore for server certificate verification.
2019-03-11T18:14:16.619+0545	INFO	20190311_122916_00007_esdqa.1.0-0-108	stderr	Mon Mar 11 18:14:16 NPT 2019 WARN: Establishing SSL connection without server's identity verification is not recommended. According to MySQL 5.5.45+, 5.6.26+ and 5.7.6+ requirements SSL connection must be established by default if explicit option isn't set. For compliance with existing applications not using SSL the verifyServerCertificate property is set to 'false'. You need either to explicitly disable SSL by setting useSSL=false, or set useSSL=true and provide truststore for server certificate verification.
2019-03-11T18:14:16.679+0545	ERROR	20190311_122916_00007_esdqa.1.0-0-108	io.prestosql.operator.Driver	Error closing operator 0 for task 20190311_122916_00007_esdqa.1.0
java.lang.NullPointerException
	at com.mysql.jdbc.MysqlIO.clearInputStream(MysqlIO.java:901)
	at com.mysql.jdbc.RowDataDynamic.close(RowDataDynamic.java:172)
	at com.mysql.jdbc.ResultSetImpl.realClose(ResultSetImpl.java:6676)
	at com.mysql.jdbc.ResultSetImpl.close(ResultSetImpl.java:851)
	at io.prestosql.plugin.jdbc.JdbcRecordCursor.close(JdbcRecordCursor.java:219)
	at io.prestosql.spi.connector.RecordPageSource.close(RecordPageSource.java:76)
	at io.prestosql.operator.TableScanOperator.finish(TableScanOperator.java:185)
	at io.prestosql.operator.TableScanOperator.close(TableScanOperator.java:174)
	at io.prestosql.operator.Driver.closeAndDestroyOperators(Driver.java:546)
	at io.prestosql.operator.Driver.processInternal(Driver.java:406)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:483)
	at io.prestosql.$gen.Presto_null__testversion____20190311_122423_1.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


2019-03-11T18:14:16.717+0545	INFO	query-execution-2	io.prestosql.event.QueryMonitor	TIMELINE: Query 20190311_122916_00007_esdqa :: Transaction:[e4d7f31b-7737-4c6e-a529-4b1710170547] :: elapsed 337ms :: planning 65ms :: scheduling 174ms :: running 81ms :: finishing 17ms :: begin 2019-03-11T18:14:16.371+05:45 :: end 2019-03-11T18:14:16.708+05:45
```

### PR DESCRIPTION
**Pull request by sopel39**
Fixes #443

### CODE REVIEW

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClientModule.java [hunk 1]
```diff
@@ -33,20 +27,6 @@
     protected void setup(Binder binder)
     {
         binder.bind(JdbcClient.class).to(MySqlClient.class).in(Scopes.SINGLETON);
-        ensureCatalogIsEmpty(buildConfigObject(BaseJdbcConfig.class).getConnectionUrl());
         configBinder(binder).bindConfig(MySqlConfig.class);
     }
-
-    private static void ensureCatalogIsEmpty(String connectionUrl)
```
Comment (findepi) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClientModule.java]: why?

Comment (sopel39) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClientModule.java]: I had to figure out another way to do so as the methods used here are gone in newest driver.

**Comment by sopel39** [comment]
Some time/timestamps tests still fail and there are still uses related to closing jdbc sink.

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 2]
```diff
@@ -127,8 +127,8 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
             throws SQLException
     {
         PreparedStatement statement = connection.prepareStatement(sql);
-        if (statement.isWrapperFor(Statement.class)) {
-            statement.unwrap(Statement.class).enableStreamingResults();
+        if (statement.isWrapperFor(StatementImpl.class)) {
```
Comment (electrum) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java]: Use the interface `com.mysql.cj.jdbc.JdbcStatement`

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClientModule.java [hunk 3]
```diff
@@ -39,14 +38,8 @@ protected void setup(Binder binder)
...
-            Properties urlProperties = driver.parseURL(connectionUrl, null);
-            checkArgument(urlProperties != null, "Invalid JDBC URL for MySQL connector");
-            checkArgument(driver.database(urlProperties) == null, "Database (catalog) must not be specified in JDBC URL for MySQL connector");
-        }
-        catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
+        checkArgument(isConnectionStringSupported(connectionUrl), "Invalid JDBC URL for MySQL connector");
+        ConnectionUrlParser parser = parseConnectionString(connectionUrl);
+        checkArgument(parser.getPath().isEmpty(), "Database (catalog) must not be specified in JDBC URL for MySQL connector");
```
Comment (electrum) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClientModule.java]: Looking at the code, `path` can be set to null. I think this should be `isNullOrEmpty(parser.getPath())`

File: presto-raptor-legacy/src/main/java/io/prestosql/plugin/raptor/legacy/util/DatabaseUtil.java [hunk 4]
```diff
@@ -109,8 +110,8 @@ public static void runIgnoringConstraintViolation(Runnable task)
     public static void enableStreamingResults(Statement statement)
             throws SQLException
     {
-        if (statement.isWrapperFor(com.mysql.jdbc.Statement.class)) {
-            statement.unwrap(com.mysql.jdbc.Statement.class).enableStreamingResults();
+        if (statement.isWrapperFor(StatementImpl.class)) {
```
Comment (electrum) [inline presto-raptor-legacy/src/main/java/io/prestosql/plugin/raptor/legacy/util/DatabaseUtil.java]: Same, use ``com.mysql.cj.jdbc.JdbcStatement``

**Comment by electrum** [comment]
We can remove `nullCatalogMeansCurrent` in `MySqlClient` since that is now the default.

**Comment by electrum** [comment]
This will need a fix for `testing-mysql-server`: https://github.com/airlift/testing-mysql-server/pull/18

**Comment by electrum** [comment]
`MySqlDataSource` in Airlift `dbpool` will need to be updated as well

**Comment by sopel39** [comment]
@electrum There are still some issues with timestamps with new driver (they differ by one day) and some other failures. I was planning to postpone work on this PR. Do you want to pick it up maybe?

**Comment by electrum** [comment]
Airlift update: https://github.com/airlift/airlift/pull/720

**Comment by electrum** [comment]
I just released `testing-mysql-server` version `8.0.12-3` that should fix this.

**Comment by findepi** [comment]
@electrum @sopel39 @kokosing  we could ensure compatibility with various MySQL versions. 
We can do this using `io.prestosql.testing.docker.DockerContainer` quite easily and run
tests on oldest and newest version supported.

**Comment by tooptoop4** [comment]
gentle ping @sopel39

**Comment by electrum** [comment]
@tooptoop4 This is on hold due to some compatibility issues to resolve. Do you have a specific need for the later version of the driver?

**Comment by tooptoop4** [comment]
i have a custom v8 mysql jar with TLS1.2 support

**Comment by hashhar** [comment]
Superseded by https://github.com/trinodb/trino/pull/7201.

That PR is functionally equivalent to this (with the review comments addressed).

**Pull request closed without merging**